### PR TITLE
fix(fieldset): manage when to show error messages

### PR DESCRIPTION
**Pull request by jorenbroekema**
for fieldsets/forms/radiogroup/checkboxgroup

- show error messages only once you leave the group
- leave fieldset/form/checkboxgroup by click outside or tab on last element
- leave radiogroup by any change of on a child radio
- some cleanup of ObserverMixin in FormatMixin and ValidateMixin

closes: #305

### CODE REVIEW

File: packages/fieldset/src/LionFieldset.js [hunk 1]
```diff
@@ -84,28 +93,46 @@ export class LionFieldset extends FormRegistrarMixin(
...
-    this.addEventListener('focused-changed', this._updateFocusedClass);
-    this.addEventListener('touched-changed', this._updateTouchedClass);
-    this.addEventListener('dirty-changed', this._updateDirtyClass);
+    this.addEventListener('dirty-changed', this._syncDirty);
+
     this._setRole();
+    document.addEventListener('click', this._checkForOutsideClick);
+  }
+
+  _checkForOutsideClick(event) {
```
Comment (CubLion) [inline packages/fieldset/src/LionFieldset.js]: what if you have 2 fieldsets, then both will be touched?

File: packages/fieldset/src/LionFieldset.js [hunk 2]
```diff
@@ -263,16 +317,20 @@ export class LionFieldset extends FormRegistrarMixin(
     }
   }
 
-  _updateFocusedClass() {
-    this.classList[this.touched ? 'add' : 'remove']('state-focused');
+  _syncFocused() {
+    this.focused = this._anyFormElementHas('focused');
```
Comment (CubLion) [inline packages/fieldset/src/LionFieldset.js]:
```suggestion
    this.focused = true;
````

File: packages/fieldset/test/lion-fieldset.test.js [hunk 3]
```diff
@@ -342,66 +332,151 @@ describe('<lion-fieldset>', () => {
...
       );
-
-      fieldset.formElements.color.dirty = true;
-      await fieldset.updateComplete;
-      expect(fieldset.classList.contains('state-dirty')).to.equal(true, 'has class "state-dirty"');
     });
 
-    it('sets prefilled when field left and value non-empty', async () => {
-      const fieldset = await fixture(html`<${tag}>${inputSlots}</${tag}>`);
+    it(`becomes "touched" once the group as a whole becomes blurred via mouse interaction after
```
Comment (CubLion) [inline packages/fieldset/test/lion-fieldset.test.js]: improve this by testing multiple fields not just one.

File: packages/validate/src/ValidateMixin.js [hunk 4]
```diff
@@ -211,30 +214,79 @@ export const ValidateMixin = dedupeMixin(
...
-          _onInfoStateChanged: ['infoState'],
-          _onSuccessStateChanged: ['successState'],
-        };
+          ].some(key => name === key)
+        ) {
+          this.validate();
+        }
+
+        // @deprecated adding css classes for backwards compatibility
+        this.constructor.validationTypes.forEach(type => {
```
Comment (CubLion) [inline packages/validate/src/ValidateMixin.js]: move the classes to async

File: packages/fieldset/src/LionFieldset.js (hunk 2 above)
Comment (daKmoR) [reply packages/fieldset/src/LionFieldset.js]: yeah won't work as the focusin event has `composed: true` 😭 
so we need this extra check here

File: packages/fieldset/test/lion-fieldset.test.js (hunk 3 above)
Comment (daKmoR) [reply packages/fieldset/test/lion-fieldset.test.js]: done

File: packages/fieldset/src/LionFieldset.js (hunk 1 above)
Comment (daKmoR) [reply packages/fieldset/src/LionFieldset.js]: done

File: packages/validate/src/ValidateMixin.js (hunk 4 above)
Comment (daKmoR) [reply packages/validate/src/ValidateMixin.js]: tried to move but fails a test in a strange way 😭 
gonna leave if for now like this... we can't save the whole world in one PR 😬

**Pull request merged**